### PR TITLE
Add support for RS485 Tx and Rx with separate IO pins

### PR DESCRIPTION
**Pull request by dhineshkumarmcci**
Added support to configure RS485 Rx with separate IO pin.
As this is a new feature update, have updated version number as v.0.5.0.

### CODE REVIEW

File: src/lib/ModbusRtu.cpp [hunk 1]
```diff
@@ -94,6 +94,9 @@ void Modbus::begin(ModbusPort *pPort, unsigned long u32speed, uint16_t config)
         // return RS485 transceiver to receive mode
         pinMode(u8txenpin, OUTPUT);
         digitalWrite(u8txenpin, LOW);
+
+        pinMode(u8rxenpin, OUTPUT);
+        digitalWrite(u8rxenpin, LOW);
     }
```
Comment (terrillmoore) [inline src/lib/ModbusRtu.cpp:101]: We are somehow missing a `Modbus::end()` function -- it's in the header file but I don't find an implementation.

Please add an `end()` function that disables both TX and RX according to the pattern, and also does an `end()` call to the data port. It should be enough to do a `Modbus::end()` to disable the com port and turn off power.

**Review by terrillmoore — CHANGES_REQUESTED** [review]
Thanks, Dhinesh!

**Comment by dhineshkumarmcci** [comment]
Have made the suggested changes, will be marking them resolved on testing the changes.

File: src/lib/ModbusRtu.cpp [hunk 2]
```diff
@@ -611,6 +625,7 @@ void Modbus::sendTxBuffer()
 
         // return RS485 transceiver to receive mode
         digitalWrite( u8txenpin, LOW );
+        digitalWrite( u8rxenpin, LOW );
```
[the remark this answers is not in the document]
Comment (dhineshkumarmcci) [reply src/lib/ModbusRtu.cpp]: will start looking at API function `set transceiver state`. 
made suggested changes to the current code

File: src/lib/ModbusRtu.cpp (hunk 1 above)
Comment (dhineshkumarmcci) [reply src/lib/ModbusRtu.cpp:103]: add method `void end()` to do `Modbus::end()`.

**Comment by dhineshkumarmcci** [comment]
tested and modbus communication works good. Though do not see any fall in power consumption with using `Modbus::end()` in the sketch

**Comment by terrillmoore** [comment]
What values are written to the GPIOs by Modbus::end()?

**Comment by dhineshkumarmcci** [comment]
In `Modbus::end()`, we write GPIOs with below value:
- Tx pin - LOW (disable Tx method used)
- Rx pin - HIGH (disable Rx method used)

**Review by terrillmoore — APPROVED** [review]
Looks good, thanks!

**Pull request merged**